File: lab2/app/src/main/java/com/example/lab2/MainActivity.java
package com.example.lab2;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.util.Log;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Iterator;
import java.util.List;
import java.util.ListIterator;

public class MainActivity extends AppCompatActivity {

    private List<Question> questionList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        loadQuestions();
        RecyclerView rv = (RecyclerView) findViewById(R.id.recyclerview);
        QuestionAdapter adapter = new QuestionAdapter(this, questionList);
        rv.setAdapter(adapter);
    }

    private void loadQuestions() {
        Log.w("loading data", "loading data");
        questionList.add(new Question(R.string.label1, R.string.text1, R.string.ac1, R.drawable.pic1));
        questionList.add(new Question(R.string.label2, R.string.text2, R.string.ac2, R.drawable.pic2));
        questionList.add(new Question(R.string.label3, R.string.text3, R.string.ac3, R.drawable.pic3));
        questionList.add(new Question(R.string.label4, R.string.text4, R.string.ac4, R.drawable.pic4));
        questionList.add(new Question(R.string.label5, R.string.text5, R.string.ac5, R.drawable.pic5));
        questionList.add(new Question(R.string.label6, R.string.text6, R.string.ac6, R.drawable.pic6));
        questionList.add(new Question(R.string.label7, R.string.text7, R.string.ac7, R.drawable.pic7));
        questionList.add(new Question(R.string.label8, R.string.text8, R.string.ac8, R.drawable.pic8));
        questionList.add(new Question(R.string.label9, R.string.text9, R.string.ac9, R.drawable.pic9));
        questionList.add(new Question(R.string.label10, R.string.text10, R.string.ac10, R.drawable.pic10));
    }
}

File: lab1/app/src/main/java/com/example/lab1t1/MainActivity.kt
package com.example.lab1t1

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.util.Log
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toDetailActivityButton: Button = findViewById(R.id.button2)

        val toUriButton: Button = findViewById(R.id.button3)

        toDetailActivityButton.setOnClickListener {
            startActivity(Intent(this, DetailActivity::class.java))
        }

        toUriButton.setOnClickListener {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/znifer")))
        }

    }

    override fun onStart() {
        super.onStart()
        Log.w("MainActivity", "MainActivity onStart() called")
    }

    override fun onResume() {
        super.onResume()
        Log.w("MainActivity", "MainActivity onResume() called")
    }

    override fun onRestart() {
        super.onRestart()
        Log.w("MainActivity", "MainActivity onRestart() called")
    }

    override fun onPause() {
        super.onPause()
        Log.w("MainActivity", "MainActivity onPause() called")
    }

    override fun onStop() {
        super.onStop()
        Log.w("MainActivity", "MainActivity onStop() called")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.w("MainActivity", "MainActivity onDestroy() called")
    }
}